### PR TITLE
Improve prover implementation

### PR DESCRIPTION
**Pull request by rakanalh**
# Description

Follow up to #857

### CODE REVIEW

File: crates/prover/src/runner.rs [hunk 1]
```diff
@@ -205,94 +202,6 @@ where
         });
     }
 
-    async fn process_l2_block(
```
Comment (rakanalh) [inline crates/prover/src/runner.rs:208]: This was just moved below `run` closer to `process_l1_block`

File: crates/prover/src/runner.rs [hunk 2]
```diff
@@ -411,9 +408,27 @@ where
...
             let mut sequencer_commitments = Vec::<SequencerCommitment>::new();
-            let mut zk_proofs = Vec::<Proof>::new();
 
-            self.extract_relevant_l1_data(l1_block, &mut sequencer_commitments, &mut zk_proofs);
+            let mut da_data = self.da_service.extract_relevant_blobs(l1_block);
+            // if we don't do this, the zk circuit can't read the sequencer commitments
+            da_data.iter_mut().for_each(|blob| {
+                blob.full_data();
+            });
+            da_data.iter_mut().for_each(|tx| {
```
Comment (rakanalh) [inline crates/prover/src/runner.rs:417]: There is no need to extract `zk_proofs` in prover as it does not process them. 

This code has been simplfied

File: crates/prover/src/runner.rs [hunk 3]
```diff
@@ -572,23 +578,21 @@ where
...
-                        {
-                            Ok(block) => block,
-                            Err(_) => {
-                                return Err(anyhow!(
-                                    "Error while fetching DA block at height: {}",
-                                    soft_batch.da_slot_height
-                                ));
-                            }
-                        };
+                    let filtered_block = match get_da_block_at_height(
```
Comment (rakanalh) [inline crates/prover/src/runner.rs:581]: This uses the `l1_block_cache` rather than fetching block through da directly.

**Review by ercecan — APPROVED** [review]

File: crates/prover/src/runner.rs [hunk 4]
```diff
@@ -411,9 +408,27 @@ where
...
+                if tx.sender().as_ref() == self.sequencer_da_pub_key.as_slice() {
+                    if let Ok(DaData::SequencerCommitment(seq_com)) = data {
+                        sequencer_commitments.push(seq_com);
+                    } else {
+                        tracing::warn!(
+                            "Found broken DA data in block 0x{}: {:?}",
+                            hex::encode(l1_block.hash()),
+                            data
+                        );
+                    }
```
Comment (eyusufatik) [inline crates/prover/src/runner.rs:429]: imo it would look better if we had a function `extract_sequencer_commitments`

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/prover/src/runner.rs (hunk 4 above)
Comment (rakanalh) [reply crates/prover/src/runner.rs]: Done

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**